app/controllers/application_controller/timelines/options.rb

```diff
@@ -36,16 +36,28 @@ def update_start_end(sdate, edate)
     def update_from_params(params)
       self.levels = params[:tl_levels]&.map(&:to_sym) || group_levels
       self.categories = {}
-      if params[:tl_categories]
-        params[:tl_categories].each do |category|
-          event_group = event_groups[events[category]].clone
-          next unless event_group.keys.include_any?(levels)
-          categories[events[category]] = {:display_name => category}
-          unless levels.include_all?(event_group.keys)
-            event_group.delete_if { |key, _v| !levels.include?(key) }
+      params.fetch(:tl_categories, []).each do |category_display_name|
+        group_data = event_groups[events[category_display_name]]
+        category = {
+          :display_name => category_display_name,
+          :include_set  => [],
+          :exclude_set  => [],
+          :regexes      => []
+        }
+
+        group_levels.each do |lvl|
+          parts = group_data[lvl]&.partition { |typ| typ.kind_of?(String) }
```

Comment:
Give the parts names, so that the later parts[0], parts[1] are readable...

```ruby
strings, regexes = group_data[lvl]&.partition { |typ| typ.kind_of?(String) }
```

Reply:
I thought `a, b = nil` would break, but I was wrong.